irohad/ametsuchi/impl/block_query_base.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BLOCK_QUERY_BASE_HPP
+#define IROHA_BLOCK_QUERY_BASE_HPP
+
+#include <soci/soci.h>
+
+#include "ametsuchi/block_query.hpp"
+#include "ametsuchi/block_storage.hpp"
+#include "logger/logger_fwd.hpp"
+
+namespace iroha::ametsuchi {
+
+  /**
+   * Class which implements BlockQuery with a Postgres backend.
+   */
+  class BlockQueryBase : public BlockQuery {
```

Comment:
Why name is BlockQueryBase? Which classes uses it as base? 

And whom are other children of BlockQuery?